### PR TITLE
mesheryctl features

### PR DESCRIPTION
**Pull request by kumarabd**
Signed-off-by: kumarabd <abishekkumar92@gmail.com>

**Description**

Feature commands added:
- mesheryctl perf apply for file
- mesheryctl mesh deploy, remove 

Fixes:
- Endpoint discovery of meshery fixed

**Notes for Reviewers**

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

File: mesheryctl/internal/cli/root/mesh/deploy.go [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+var (
+	deployCmd = &cobra.Command{
+		Use:   "deploy",
+		Short: "deploy a service mesh in the kubernetes cluster",
+		Args:  cobra.MinimumNArgs(1),
+		Long:  `deploy service mesh in the connected kubernetes cluster`,
+		PreRunE: func(cmd *cobra.Command, args []string) error {
+			log.Infof("Verifying prerequisites...")
+			if len(args) < 1 {
+				return errors.New("no service mesh specified")
```
Comment (leecalcote) [inline mesheryctl/internal/cli/root/mesh/deploy.go:27]: Will we be showing them a list of available service meshes to provision based on the connected adapters?

File: mesheryctl/internal/cli/root/mesh/deploy.go [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+			}
+
+			mctlCfg, err := config.GetMesheryCtl(viper.GetViper())
+			if err != nil {
+				return errors.Wrap(err, "error processing config")
+			}
+
+			if err = validateAdapter(mctlCfg, tokenPath, args[0]); err != nil {
+				return errors.Wrap(err, "adapter not valid")
```
Comment (leecalcote) [inline mesheryctl/internal/cli/root/mesh/deploy.go:36]:
```suggestion
				return errors.Wrap(err, "unable to find adapter. Please check its spelling and port number")
```

File: mesheryctl/internal/cli/root/mesh/deploy.go [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+
+func validateAdapter(mctlCfg *config.MesheryCtlConfig, tokenPath string, name string) error {
+	prefs, err := utils.GetSessionData(mctlCfg, tokenPath)
+	if err != nil {
+		return err
+	}
+
+	adapterNames := []string{}
+	for _, adapter := range prefs.MeshAdapters {
+		if adapter.Name == name || adapter.Name == strings.ToUpper(name) {
```
Comment (leecalcote) [inline mesheryctl/internal/cli/root/mesh/deploy.go:151]: Is this accounting for the adapter's port number as well?

File: mesheryctl/internal/cli/root/mesh/deploy.go (hunk 1 above)
Comment (kumarabd) [reply mesheryctl/internal/cli/root/mesh/deploy.go:27]: Covering this in upcoming PR.

**Pull request merged**